src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/posix_actor.cc

```diff
@@ -20,14 +23,52 @@ int PosixActor::Stop() {
     LAVA_LOG_ERR("process %d waitpid error\n", this->pid_);
     return -1;
   }
+  this->status_ = StatsStopped;
 
   // Check the status
   return 0;
 
 }
 
-int PosixActor::Wait() {
+int PosixActor::Stop() {
+  int status;
+  kill(pid_, SIGTERM);
```

Comment:
we need to send `stop` cmd/signal to target actor, not kill.
maybe rename as `ForceStop`.